openedx/frontend-app-learner-record
# feat: add plugin slot to Footer using FPF

### Description
This PR installs the master branch of Frontend Plugin Framework library into Learner Record. The primary goal of this PR is:
- confirm FPF can be installed in an MFE and not break anything at Prod level
- confirm FPF can wrap a Plugin Slot around a React dependency (frontend-component-footer) and insert an arbitrary plugin in the same slot

This PR also includes:
- update frontend-build patch to remove `require` statement, which prevented any `import` in the `env.config.js`. 

### To test this locally:
1. Run `npm install`
2. At the root directory, create an `env.config.jsx` file that includes the following: 
```
import { PLUGIN_OPERATIONS, DIRECT_PLUGIN } from '@edx/frontend-plugin-framework/src';

const config = {
  pluginSlots: {
    footer_plugin_slot: {
      keepDefault: true,
      plugins: [
        {
          op: PLUGIN_OPERATIONS.Insert,
          widget: {
            id: 'hidden_plugin',
            type: DIRECT_PLUGIN,
            priority: 60,
            RenderWidget: () => (
              <div className="empty-plugin" />
            ),
          },
        },
      ],
    },
  },
};

export default config;
```

3. Run `npm run start`
4. Go to https://localhost:1990
5. An empty `<div>` should be underneath the `<Footer>` tag when you inspect the source code. 

#### Ways to play around with this
1. Add a `<p>` inside the `div` that shows some text.
2. Set `keepDefault` to `false` in JS config to hide Footer and still insert empty `div`
3. Remove `env.config.jsx` completely to ensure Footer isn't affected by a missing config (note: `keepDefault` defaults to `true` on FPF's side, which is why the Footer is still present without the JS config)

### Screenshots
#### Insert empty `div` plugin
![Screenshot 2024-03-14 at 1 18 49 PM](https://github.com/openedx/frontend-app-learner-record/assets/62807795/5a5a233b-bf95-45a7-9b1d-10907571523a)

#### Remove `env.config.jsx` entirely
![Screenshot 2024-03-14 at 1 21 02 PM](https://github.com/openedx/frontend-app-learner-record/assets/62807795/185cbc89-71ea-426a-a3d5-cceb0eb0c19e)

[APER-3275](https://2u-internal.atlassian.net/browse/APER-3275)
